src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectWithParameterizedConstructorConverter.cs

```diff
@@ -23,6 +23,14 @@ internal abstract partial class ObjectWithParameterizedConstructorConverter<T> :
 
         internal sealed override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options, ref ReadStack state, [MaybeNullWhen(false)] out T value)
         {
+            JsonTypeInfo jsonTypeInfo = state.Current.JsonTypeInfo;
```

Comment:
I think you might need similar fix in LargeObjectWithParametrizedConsturctor (or whatever it's called) - you might want to copy&paste the tests and add at least 4-5 args to hit that code path